server/src/main/java/fr/uga/l3miage/tp4/models/UserEntity.java

```diff
@@ -0,0 +1,19 @@
+package fr.uga.l3miage.tp4.models;
+
+import javax.persistence.*;
+
+@Entity
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+public abstract class UserEntity {
```

Comment:
il manque ton `@MappedSuperclass` , et attention le `@Entity` c'est soit sur les entités filles soit juste sur l'entités mère
```suggestion
@Entity
@MappedSuperclass
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
public abstract class UserEntity {
```